ruby/ql/lib/codeql/ruby/controlflow/internal/Completion.qll

```diff
@@ -310,7 +310,7 @@ class MatchingCompletion extends ConditionalCompletion {
     this = TNestedCompletion(_, TMatchingCompletion(value), _)
   }
 
-  override SuccessorType getAMatchingSuccessorType() {
+  override ConditionalSuccessor getAMatchingSuccessorType() {
```

Comment:
Could you change it to `MatchingSuccessor` instead and remove the infix cast below?

Reply:
That does not work for `class NestedMatchingCompletion extends NestedCompletion, MatchingCompletion`, does it?

Reply:
`Completion`s are different from `SuccessorType`s (the latter are never nested).

Reply:
What I meant is that `override BooleanSuccessor getAMatchingSuccessorType()` defined in `NestedMatchingCompletion` would not match the signature if we restricted it to `MatchingSuccessor` in the super class (`MatchingCompletion`)

Reply:
Ah, right, sorry.